autogpt_platform/backend/backend/blocks/llm.py

```diff
@@ -29,7 +30,21 @@ class ModelMetadata(NamedTuple):
     cost_factor: int
 
 
-class LlmModel(str, Enum):
+class LlmModelMeta(EnumMeta):
+    @property
+    def __members__(cls):
+        if Settings().config.behave_as == BehaveAs.LOCAL:
+            return super().__members__
+        else:
+            removed_providers = ["ollama"]
+            return {
+                name: member
+                for name, member in super().__members__.items()
+                if LlmModel[name].provider not in removed_providers
+            }
```

Comment:
any logic could be done here, but we should try and make sure it's fast since it will run any time the class member list is accessed. 

We could load the enables from a file etc if desired